substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/serialize/hosted/SerializationFeature.java

```diff
@@ -96,13 +121,140 @@ public void duringSetup(DuringSetupAccess a) {
                         ConfigurationFile.SERIALIZATION.getFileName());
     }
 
+    private static GraphBuilderConfiguration buildLambdaParserConfig(ClassInitializationPlugin cip) {
+        GraphBuilderConfiguration.Plugins plugins = new GraphBuilderConfiguration.Plugins(new InvocationPlugins());
+        plugins.setClassInitializationPlugin(cip);
+        return GraphBuilderConfiguration.getDefault(plugins).withEagerResolving(true);
+    }
+
+    @SuppressWarnings("try")
+    private static StructuredGraph createMethodGraph(ResolvedJavaMethod method, DebugContext debug) {
+        StructuredGraph graph = new StructuredGraph.Builder(debug.getOptions(), debug).method(method).build();
+        try (DebugContext.Scope ignored = debug.scope("ParsingToMaterializeLambdas")) {
+            GraphBuilderPhase lambdaParserPhase = new GraphBuilderPhase(buildLambdaParserConfig(new NoClassInitializationPlugin()));
+            HighTierContext context = new HighTierContext(GraalAccess.getOriginalProviders(), null, OptimisticOptimizations.NONE);
+            lambdaParserPhase.apply(graph, context);
+        } catch (Throwable e) {
+            throw debug.handle(e);
+        }
+        return graph;
+    }
+
+    private static Class<?> lookupForMemberField(Class<?> clazz, Object constantObject) {
```

Comment:
I would try to avoid reflection here. How does the `org.graalvm.compiler.nodes.java.NewInstanceNode#instanceClass` get the lambda type. I doubt that the compiler does reflection of this sort.

Reply:
It does not. But `org.graalvm.compiler.nodes.java.NewInstanceNode` is being created during parsing the specific method. In that case, name of the `instanceClass` is got from the constant pool using `CPI` inside the `org.graalvm.compiler.java.BytecodeParser`. The only way that we can do this for `ConstantNodes` is to try to change parsing process, which I think is not a good idea.